### PR TITLE
custom_db_role resource use *string

### PR DESCRIPTION
**Pull request by Freyert**
## Description

Fixes https://github.com/mongodb/go-client-mongodb-atlas/issues/263

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code

## Further comments

Just posting this here to streamline any fix on this. If further doc/tests/etc. are needed let me know. Didn't want to run the tests in case they were integration and not just unit tests.

I did use a `replace` in the `go.mod` to use my fork and this fixes my issue, but you have to wrap all strings as pointers. Therefore, this is a breaking change.

### CODE REVIEW

**Review by themantissa — DISMISSED** [review]
LGTM.  Added DoU so one can review so we can figure out the impact downstream.  TY!

**Comment by Freyert** [comment]
I reworked the test to loop over a list of cases so it is easier to add more cases. I also added an extra case to ensure when we use "cluster" resources it works correctly.

File: mongodbatlas/custom_db_roles.go [hunk 1]
```diff
@@ -41,9 +41,9 @@ var _ CustomDBRolesService = &CustomDBRolesServiceOp{}
 
 // A Resource describes a specific resource the Role will allow operating on.
 type Resource struct {
-	Collection string `json:"collection,omitempty"`
-	Db         string `json:"db,omitempty"` //nolint:stylecheck // not changing this as is a breaking change
-	Cluster    *bool  `json:"cluster,omitempty"`
+	Collection *string `json:"collection,omitempty"`
+	Db         *string `json:"db,omitempty"` //nolint:stylecheck // not changing this as is a breaking change
```
[the remark this answers is not in the document]
Comment (Freyert) [reply mongodbatlas/custom_db_roles.go]: Just added this eb1bc7a (misunderstood it the first time)

File: mongodbatlas/custom_db_roles_test.go [hunk 2]
```diff
@@ -42,8 +42,8 @@ func TestCustomDBRoles_ListCustomDBRoles(t *testing.T) {
 		Actions: []Action{{
 			Action: "CREATE_INDEX",
 			Resources: []Resource{{
-				Collection: "test-collection",
-				Db:         "test-db",
+				Collection: StrPtr("test-collection"),
```
[the remark this answers is not in the document]
Comment (Freyert) [reply mongodbatlas/custom_db_roles_test.go]: dfa4844

**Review by gssbzn — APPROVED** [review]
LGTM thanks for all the changes here

**Review by coderGo93 — APPROVED** [review]
LGTM

**Review by themantissa — APPROVED** [review]

**Comment by themantissa** [comment]
@gssbzn  we are okay for this to be merged.  We'll address it in the provider.

**Pull request merged**